File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/core/exchanger/SwlExchangerConfig.js

/**
 * @return {SwlExchangerConfig}
 * @constructor
 */
function SwlExchangerConfig() {
    /**
     *
     * @type {boolean}
     */
    this.enableTimestamp = true;
    /**
     *
     * @type {long}
     */
    this.timestampExpireWindowSeconds = 30;

    /**
     *
     * @type {boolean}
     */
    this.enableNonce = false;
    /**
     *
     * @type {long}
     */
    this.nonceTimeoutSeconds = 180;

    /**
     *
     * @type {boolean}
     */
    this.enableEncrypt=true;
    /**
     *
     * @type {boolean}
     */
    this.enableDigital = true;
}

export default SwlExchangerConfig

File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/core/exchanger/SwlExchangerConfig.java
package i2f.swl.core.exchanger;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.concurrent.TimeUnit;

/**
 * @author Ice2Faith
 * @date 2024/7/10 17:34
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlExchangerConfig {
    protected boolean enableTimestamp = true;
    protected long timestampExpireWindowSeconds = 30;

    protected boolean enableNonce = false;
    protected long nonceTimeoutSeconds = TimeUnit.MINUTES.toSeconds(30);

    protected boolean enableEncrypt = true;

    protected boolean enableDigital = true;
}
